### PR TITLE
Implement post editing functionality

### LINKED ISSUE #124: Implement post editing functionality
Editing a post is considered important basic functionality. Let's add it to the library before the 2.0.0 release.

Link to the API call:
https://docs.joinmastodon.org/methods/statuses/#edit

### PR DESCRIPTION
**Pull request by bocops**
Implements
- post editing in separate methods `editStatus()` and `editPoll()`, comparable to existing `post*()` methods
- getting edit history in `getEditHistory()`
- getting non-HTML strings to edit them in `getStatusSource()`

Adds Rx methods and tests.

Closes #124.

### CODE REVIEW

File: bigbone/src/main/kotlin/social/bigbone/api/entity/StatusEdit.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package social.bigbone.api.entity
+
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Represents a revision of a status that has been edited.
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/StatusEdit.kt:6]: A link to the Mastodon documentation would be nice.

File: bigbone/src/main/kotlin/social/bigbone/api/method/StatusMethods.kt [hunk 2]
```diff
@@ -421,4 +423,108 @@ class StatusMethods(private val client: MastodonClient) {
...
+                mediaIds?.let { append("media_ids", it) }
+                append("sensitive", sensitive)
+                spoilerText?.let { append("spoiler_text", it) }
+                language?.let { append("language", it) }
+            }
+        )
+    }
+
+    /**
+     * Edit a given status to change its text, sensitivity, or poll. Note that editing a poll’s options will reset the votes.
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/method/StatusMethods.kt:462]: Typo: Should be "Edit a given poll..."

**Review by andregasser — APPROVED** [review]
This looks great, thx! Just had some minor comments. Please have a look. Otherwise good to merge 🙂

File: bigbone/src/main/kotlin/social/bigbone/api/method/StatusMethods.kt (hunk 2 above)
Comment (bocops) [reply bigbone/src/main/kotlin/social/bigbone/api/method/StatusMethods.kt:462]: As discussed externally, the distinction between "polls" and "statuses" is one of our library - Mastodon considers both of them to be statuses. As such, it is possible to edit a status with poll using `editStatus()` if we don't want to touch its poll data, or potentially to edit a status _without_ poll using `editPoll()` to add one.

**Review by andregasser — APPROVED** [review]

**Pull request merged**